tests/testthat/test-carehomes-multistrain.R

```diff
@@ -1012,50 +1012,100 @@ test_that("Stuck when gamma =  0", {
 })
 
 
+
 test_that("Stuck when gamma =  0 for second strain", {
   np <- 3L
 
-  ## gammaP is 0 so IC1 is 0 for second strain
+  ## gammaP is 0 so IC1 is 0
   p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
                             strain_transmission = c(1, 1),
-                            strain_rel_gamma_A = c(1, 1),
+                            strain_rel_gamma_A = 1,
                             strain_rel_gamma_P = c(1, 0),
-                            strain_rel_gamma_C_1 = c(1, 1),
-                            strain_rel_gamma_C_2 = c(1, 1),
+                            strain_rel_gamma_C_1 = 1,
+                            strain_rel_gamma_C_2 = 1,
                             strain_seed_date =
-                              c(sircovid_date("2020-02-07"),
-                                sircovid_date("2020-02-08")),
+                              rep(sircovid_date("2020-02-07"), 2),
                             strain_seed_rate = c(10, 0))
+  p$gamma_P = c(0, 1)
 
   mod <- carehomes$new(p, 0, np, seed = 1L)
 
   initial <- carehomes_initial(mod$info(), 10, p)
   mod$set_state(initial$state, initial$step)
 
   index_I_A <- mod$info()$index$I_A
-  index_I_A_strain_2 <- index_I_A[20:38]
-  index_I_A_strain_1 <- index_I_A[1:19]
   index_I_P <- mod$info()$index$I_P
-  index_I_P_strain_1 <- index_I_P[1:19]
-  index_I_P_strain_2 <- index_I_P[20:38]
   index_I_C_1 <- mod$info()$index$I_C_1
-  index_I_C_1_strain_1 <- index_I_C_1[1:19]
-  index_I_C_1_strain_2 <- index_I_C_1[20:38]
   index_I_C_2 <- mod$info()$index$I_C_2
-  index_I_C_2_strain_1 <- index_I_C_2[1:19]
-  index_I_C_2_strain_2 <- index_I_C_2[20:38]
 
   end <- sircovid_date("2020-05-01") / p$dt
   steps <- seq(initial$step, end, by = 1 / p$dt)
 
   set.seed(1)
   y <- mod$simulate(steps)
-  expect_equal(p$gamma_P[2], 0)
-  expect_false(all(unlist(y[index_I_C_1_strain_1, , ]) == 0))
-  expect_true(all(unlist(y[index_I_C_1_strain_2, , ]) == 0))
-  expect_false(all(unlist(y[index_I_P_strain_2, , ]) == 0))
+  expect_true(all(unlist(y[index_I_C_1, , ]) == 0))
```

Comment:
I think the previous test was better because it was able to test that you could stop progression for one strain but not the other